src/utils.cpp

```diff
@@ -47,4 +50,54 @@ namespace utils {
       }
     }
   }
+
+  void run_interruptible(const std::function<void()>& exec_fn, const std::function<void()>& cleanup_fn)
+  {
+    std::exception_ptr eptr;
+#if !defined(_WIN32) && !defined(_WIN64)
+    sigset_t set, old_set;
+    sigemptyset(&set);
+    sigaddset(&set, SIGINT);
+    int rc = pthread_sigmask(SIG_BLOCK, &set, &old_set);
+    if ( rc != 0 )
+    {
+      Rcpp::warning("Unable to properly mask SIGINT from execution thread.");
```

Comment:
```suggestion
      // Unable to properly mask SIGINT from execution thread.
      Rcpp::warning("<a more user-facing message>");
```

What does this mean for the user? I'm wondering if this would be better situated as a message, error, or just a code comment.

If this remains a warning, could we `pretty_print_message()` it?

Reply:
Done/thanks